sympy/solvers/tests/test_solveset.py

```diff
@@ -1238,13 +1236,13 @@ def test_trig_system():
     # TODO: add more simple testcases when solveset returns
     # simplified soln for Trig eq
     assert nonlinsolve([sin(x) - 1, cos(x) -1 ], x) == S.EmptySet
-    soln1 = (ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),)
-    soln = FiniteSet(soln1)
-    assert nonlinsolve([sin(x) - 1, cos(x)], x) == soln
 
 
 @XFAIL
 def test_trig_system_fail():
+    soln1 = (ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers),)
+    soln = FiniteSet(soln1)
+    assert nonlinsolve([sin(x) - 1, cos(x)], x) == soln
```

Comment:
I dont know why this test is giving wrong solution but i debugged it with python debugger, the expression entering in ImageSet Union are giving correct result. Something else is giving wrong answer. I will try to debug it. Rest all of the tests are giving correct result.